ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallOptions.kt

```diff
@@ -60,6 +60,30 @@ class PaywallOptions(builder: Builder) {
             .build()
     }
 
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is PaywallOptions) return false
+
+        if (offeringSelection != other.offeringSelection) return false
+        if (shouldDisplayDismissButton != other.shouldDisplayDismissButton) return false
+        if (fontProvider != other.fontProvider) return false
+        if (listener != other.listener) return false
+        if (mode != other.mode) return false
+        if (dismissRequest != other.dismissRequest) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
```

Comment:
If we made `PaywallOptions` a `data class` would we get this for free?

Reply:
Though my other suggestion might remove the need for this too.

Reply:
> If we made PaywallOptions a data class would we get this for free?

I was trying to avoid it, since it's not recommended to expose data classes... But honestly, it's reached a point where things are much easier, so I think I will do the change yeah.

Reply:
Made the change to data classes https://github.com/RevenueCat/purchases-android/pull/1417/commits/e262e9af760c79fd29776ccd00700132bdf61852

Reply:
👍🏻 
Why is it not recommended?

Reply:
Mentioned it over slack but this article has some of the drawbacks: https://jakewharton.com/public-api-challenges-in-kotlin/